packages/ccd-js-gen/src/lib.ts

```diff
@@ -0,0 +1,275 @@
+            name: contractClassId,
+            properties: [
+                {
+                    docs: [
+                        'Name of the smart contract supported by this client.',
+                    ],
+                    scope: tsm.Scope.Public,
+                    isReadonly: true,
+                    name: contractNameId,
+                    type: 'string',
+                    initializer: `'${contract.contractName}'`,
+                },
+                {
+                    docs: ['Generic contract client used internally.'],
+                    scope: tsm.Scope.Private,
+                    name: genericContractId,
+                    type: 'SDK.Contract',
+                },
+                {
+                    docs: ['Dry run entrypoints of the smart contract.'],
+                    scope: tsm.Scope.Public,
+                    name: dryRunId,
+                    type: contractDryRunClassId,
+                },
+            ],
+        });
+
+        const dryRunClassDecl = sourceFile.addClass({
+            docs: [
+                `Smart contract client for dry running messages to a contract instance of '${contract.contractName}' on chain.`,
+            ],
+            isExported: true,
+            name: contractDryRunClassId,
+        });
+
+        classDecl
+            .addConstructor({
+                docs: ['Contruct a client for a contract instance on chain'],
+                parameters: [
+                    {
+                        name: grpcClientId,
+                        type: 'SDK.ConcordiumGRPCClient',
+                        scope: tsm.Scope.Public,
+                    },
+                    {
+                        name: contractAddressId,
+                        type: 'SDK.ContractAddress',
+                        isReadonly: true,
+                        scope: tsm.Scope.Public,
+                    },
+                ],
+            })
+            .setBodyText(
+                `this.${genericContractId} = new SDK.Contract(${grpcClientId}, ${contractAddressId}, '${contract.contractName}');
+this.${dryRunId} = new ${contractDryRunClassId}(this.${genericContractId});`
+            );
+
+        dryRunClassDecl.addConstructor({
+            docs: ['Contruct a client for a contract instance on chain'],
+            parameters: [
+                {
+                    name: genericContractId,
+                    type: 'SDK.Contract',
+                    scope: tsm.Scope.Private,
+                },
+            ],
+        });
+
+        for (const entrypointName of contract.entrypointNames) {
+            const transactionMetadataId = 'transactionMetadata';
+            const parameterId = 'parameter';
+            const signerId = 'signer';
+            classDecl
+                .addMethod({
+                    docs: [
+                        `Send a message to the '${entrypointName}' entrypoint of the '${contract.contractName}' contract.`,
+                    ],
```

Comment:
And correspondingly also here.

Also, what happens if you try to send a view function invocation as a contract update? Should we explain somewhere that this is a limitation of the current implementation?

Reply:
Just to answer your question: The chain and smart contract is not really aware of 'view functions'. A receive-function of a smart contract have the option to return a value, and when invoked using `invokeInstance` it will run the code locally just in the node and provide you this return value, which is then used for something like view functions. Sending a contract update transaction to the chain will just ignore the return value.
In other words, sending update transactions to view functions will be treated as any other receive function, but since a view function usually don't do any side-effects it will just cost the caller some CCD.